tests/foreman/api/test_provisioning.py

```diff
@@ -0,0 +1,136 @@
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+import pytest
+from fauxfactory import gen_string
+from packaging.version import Version
+from wait_for import wait_for
+
+
+@pytest.mark.on_premises_provisioning
+@pytest.mark.rhel_ver_match('[^6]')
+def test_rhel_pxe_provisioning(
+    request,
+    module_provisioning_sat,
+    module_org_with_manifest,
+    module_location,
+    provisioning_host,
+    module_provisioning_rhel_content,
+    module_lce_library,
+    module_default_org_view,
+):
+    """Simulate baremetal provisioning of a RHEL system via PXE on RHV provider
+
+    :id: 8b33f545-c4a8-428d-8fd8-a5e402c8cd10
+
+    :steps:
+        1. Provision RHEL system via PXE on RHV
+        2. Check that resulting host is registered to Satellite
+        3. Check host is subscribed to Satellite
+
+    :expectedresults:
+        1. Host installs right version of RHEL
+        2. Satellite is able to run REX job on the host
+        3. Host is registered to Satellite and subscription status is 'Success'
+    """
+    host_mac_addr = provisioning_host._broker_args['provisioning_nic_mac_addr']
+    sat = module_provisioning_sat.sat
+
+    host = sat.api.Host(
+        hostgroup=module_provisioning_sat.hostgroup,
+        organization=module_org_with_manifest,
+        location=module_location,
+        content_facet_attributes={
+            'content_view_id': module_default_org_view.id,
+            'lifecycle_environment_id': module_lce_library.id,
+        },
+        name=gen_string('alpha').lower(),
+        mac=host_mac_addr,
+        operatingsystem=module_provisioning_rhel_content.os,
+        subnet=module_provisioning_sat.subnet,
+        host_parameters_attributes=[
+            {'name': 'remote_execution_connect_by_ip', 'value': 'true', 'parameter_type': 'boolean'}
+        ],
+        build=True,  # put the host in build mode
+    ).create(create_missing=False)
+    # Clean up the host to free IP leases on Satellite.
+    # broker should do that as a part of the teardown, putting here just to make sure.
+    request.addfinalizer(host.delete)
+
+    # Start the VM, do not ensure that we can connect to SSHD
+    provisioning_host.power_control(ensure=False)
+
+    # TODO: Implement Satellite log capturing logic to verify that
+    # all the events are captured in the logs.
+
+    # Host should do call back to the Satellite reporting
+    # the result of the installation. Wait until Satellite reports that the host is installed.
+    wait_for(
+        lambda: host.read().build_status_label != 'Pending installation',
+        timeout=1500,
+        delay=10,
+    )
+    host = host.read()
+    assert host.build_status_label == 'Installed'
+
+    # Change the hostname of the host as we know it already.
+    # In the current infra environment we do not support
+    # addressing hosts using FQDNs, falling back to IP.
+    provisioning_host.hostname = host.ip
+    # Host is not blank anymore
+    provisioning_host.blank = False
```

Comment:
shouldn't this action exist in the tear down section of the `provisioning_host` fixture? I understand it is required only for the teardown (checkin) to succeed and has nothing to do with the test itself